app/models/office.rb

```diff
@@ -115,4 +131,28 @@ def self.scope_by_office_users_if_coordinator(scope, user)
       raise 'You do not have the proper permissions'
     end
   end
+
+  def self.percent_satisfied_by_office(offices)
+    offices.reduce([]) do |arr, office|
+      arr << [office.name, needs_satisfied_percent(office.needs)]
+    end
+  end
+
+  def self.percent_satisfied_by_state(offices)
+    offices.group_by { |o| o.address.state }.map do |state, offices_by_state|
+      ar_collection_needs = Need.where(id: offices_by_state.map(&:needs).flatten.map(&:id))
+
+      [state, needs_satisfied_percent(ar_collection_needs)]
+    end
+  end
+
+  def self.percent_satisfied_total(offices)
+    ar_collection_needs = Need.where(id: offices.map(&:needs).flatten.map(&:id))
```

Comment:
```suggestion
    ar_collection_needs = Need.where(office_id: offices.pluck(:id))
```

Does this yield the same result?

Reply:
Yes!